### PR TITLE
Support FEP-f228 activity collection backfill

### LINKED ISSUE #275: `@fedify/backfill` package for FEP-f228 conversation backfill
The “quiet fediverse” problem occurs when users see incomplete conversations due to the distributed nature of ActivityPub. Currently, most implementations use reply tree crawling (recursively fetching `inReplyTo` chains), which can be inefficient and may miss replies.

[FEP-f228: Backfilling conversations](https://w3id.org/fep/f228) proposes a more efficient approach where conversation owners provide resolvable `context` properties that resolve to `OrderedCollection`s containing the entire conversation thread.

## Proposal

Create a new `@fedify/backfill` package that implements conversation backfilling with multiple strategies:

1. *FEP-f228 context owner approach* (primary): Check if `context` resolves to an `OrderedCollection`
2. *Reply tree crawling* (fallback): Traditional recursive fetching via `inReplyTo`
3. *Hybrid approach*: Combine both methods for maximum coverage

```ts
import { backfill } from "@fedify/backfill";

const conversation = await Array.fromAsync(
  backfill(context, note, {
    strategies: ["context-owner", "reply-tree-crawl"],
  })
);
```

## Use cases

- ActivityPub servers wanting to display complete conversations
- Clients needing to backfill missing replies
- Bridge implementations between different fediverse platforms

## Implementation considerations

- Should work with any Fedify-based application
- Configurable strategies and fallback mechanisms  
- Efficient deduplication and caching
- Rate limiting and error handling for external fetches

## Related

- https://github.com/TryGhost/ActivityPub/issues/901
- [Backfilling Conversations: Two Major Approaches](https://community.nodebb.org/topic/18844/backfilling-conversations-two-major-approaches)

### PR DESCRIPTION
**Pull request by sij411**
## Summary

This PR adds the next FEP-f228 context collection path for `@fedify/backfill`: context collections whose items are Activity objects.

The new strategy handling keeps `BackfillItem.object` focused on conversation objects. When a supported `Create` activity is found, backfill extracts and yields the activity's object rather than yielding the activity itself.

Related to #275 

## Included

- Add `context-auto`, `context-objects`, and `context-activities` strategy handling
- Make `context-auto` the default classifier for direct objects and supported activities
- Extract embedded or dereferenced `Create.object` values from activity collections
- Reuse existing request budgets, abort handling, interval handling, and dedupe behavior
- Add tests for explicit, automatic, empty, overlapping, duplicate, embedded, URL, and budget cases
- Document the strategy behavior in `@fedify/backfill`

## Validation

- `deno task -f @fedify/backfill check`
- `deno task -f @fedify/backfill test`
- `pnpm --filter @fedify/backfill test`

AI assistance disclosed in commit message with `Assisted-by: Codex:gpt-5`.

### CODE REVIEW

**Comment by sij411** [comment]
@gemini review

**Comment by sij411** [comment]
@gemini review

**Comment by sij411** [comment]
@coderabbitai review

**Comment by dahlia** [comment]
It's trivial, but I think the `Assisted-by` trailer on your commit should be:

```
Assisted-by: Codex:gpt-5.5
```

… instead of:

```
Assisted-by: Codex:gpt-5
```

**Review by dahlia — APPROVED** [review]

File: packages/backfill/src/types.ts [hunk 1]
```diff
@@ -5,7 +5,10 @@ import type { Object as APObject } from "@fedify/vocab";
  *
  * @since 2.x.0
  */
-export type BackfillStrategy = "context-posts";
+export type BackfillStrategy =
```
Comment (2chanhaeng) [inline packages/backfill/src/types.ts:16]: It would be good to have a description for each item. Especially, it would be great to include the fact that including `"context-auto"` in `BackfillOptions.strategies` absorbs all other strategies.

File: packages/backfill/src/types.ts [hunk 2]
```diff
@@ -56,6 +59,15 @@ export interface BackfillContext {
...
   TObject extends APObject = APObject,
 > {
+  /**
+   * Backfill strategies to run.
+   *
+   * Defaults to `["context-auto"]`.
+   *
+   * @since 2.x.0
+   */
+  readonly strategies?: readonly BackfillStrategy[];
```
Comment (2chanhaeng) [inline packages/backfill/src/types.ts:78]: It would be great to include the fact that including `"context-auto"` in `BackfillOptions.strategies` absorbs all other strategies at this part, too.

File: packages/backfill/README.md [hunk 3]
```diff
@@ -61,3 +61,23 @@ for await (
 
 The seed object itself is not yielded.  If it appears in the discovered
 collection, it is skipped by ID.
+
+By default, `backfill()` uses the `context-auto` strategy.  In this mode,
+collection items are treated as backfillable objects by default.  If an item is
+recognized as an Activity, `backfill()` extracts the activity's object instead.
```
Comment (2chanhaeng) [inline packages/backfill/README.md]: By *packages/backfill/src/backfill.ts:174*, the `context-auto` strategy processes only `Create`, like the `context-activities` strategy. But this doc says *"Activity"*, which makes it seem like the strategy processes all kinds of Activities.

**Review by 2chanhaeng — CHANGES_REQUESTED** [review]
The code is great! It would be better if some docs and comments were fixed.

**Comment by sij411** [comment]
Doc nitpicks are addressed in [9776f2d](https://github.com/fedify-dev/fedify/pull/801/commits/9776f2d63cfe74dd1421aa2fe44eaacd15348610)

**Review by 2chanhaeng — APPROVED** [review]
👍👍👍
